lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -978,6 +1001,26 @@ function dosomething_user_get_user_by_email($email) {
 }
 
 /**
+ * Search for users users by their northstar id.
+ *
+ * @param string $nid
+ *   The northstar id for which you are searching.
+ *
+ * @return string/bool
+ *   The user's id in phoenix, or false if no user was found.
+ */
+function dosomething_user_get_user_by_northstar_id($id) {
+  $query = db_select('field_data_field_northstar_id', 'f')
+          ->condition('f.field_northstar_id_value', $id)
+          ->fields('f', ['entity_id']);
+  $result = $query->execute()->fetchField();
```

Comment:
so nitpicky, but I would save this var as `uid` so when you call `user_load` on it it doesn't look as ~wild~